rosbag2_transport/test/rosbag2_transport/mock_sequential_reader.hpp

```diff
@@ -31,6 +31,7 @@ class MockSequentialReader : public rosbag2_cpp::reader_interfaces::BaseReaderIn
   {
     (void) storage_options;
     (void) converter_options;
+    num_read_ = 0;
```

Comment:
Remove unrelated change

Reply:
Actually the MockReader needs this in order to be able to read the messages from the beginning when it has read all messages. Otherwise the has_next() method would return false after the first iteration in the loop and the MockReader wouldn't be able to reread the first message.